index.js

```diff
@@ -126,7 +126,9 @@ function replacePath (path, keys) {
       name: key || index++,
       delimiter: prefix || '/',
       optional: optional,
-      repeat: repeat
+      repeat: repeat,
+      offset: i + (prefix ? prefix.length : 0),
```

Comment:
Ok, I'm thinking tokens might be a better approach now. Since we need to remove escape characters from the output and I think it makes sense to provide as much help to the user as possible. Thoughts?